server/index.js

```diff
@@ -72,6 +72,106 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+// below is for /api/cart
+
+app.get('/api/cart', (req, res, next) => {
+  // gets all productsin cart
+  if (!req.session.cartId) {
+    return res.json([]);
+  }
+  const sql = `
+    select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+      where "c"."cartId" = $1
+  `;
+  db.query(sql, [req.session.cartId])
+    .then(result => {
+      const products = result.rows;
+      res.json(products);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
```

Comment:
You can just pass the error to `next()` here. The final middleware will send the `500` response for you.
```js
.catch(err => next(err));
```